internal/args/parse.go

```diff
@@ -11,15 +11,29 @@ import (
 
 const docsURL = "https://github.com/release-engineering/exodus-rsync"
 
-type filterArgument string
-
-func (f filterArgument) Validate() error {
-	if f == "+ */" {
-		// This is OK as it means nothing is filtered
-		return nil
+type filterArguments []string
+
+func (f filterArguments) Validate() error {
+	validRules := []string{"+", "-"}
+	validMods := []string{"", "/"}
+	validSeps := []string{" ", "_"}
+validation:
+	for _, arg := range f {
+		arg = strings.Trim(arg, "'")
```

Comment:
I don't understand this part, why are we trimming `'` ?

rsync itself would reject filters with extraneous `'` characters, e.g.

```
$ rsync -avz ./src ./dest --filter "+ *"
sending incremental file list

sent 96 bytes  received 17 bytes  226.00 bytes/sec
total size is 0  speedup is 0.00

$ rsync -avz ./src ./dest --filter "'+ *'"
Unknown filter rule: `'+ *''
rsync error: syntax or usage error (code 1) at exclude.c(919) [client=3.2.3]
```

Reply:
It's a workaround. The parser won't accept arguments like "--filter - *", it'll read "- *" as a flag.
`rsync: error: --filter: expected string value but got "- *" (short flag); perhaps try --filter="- *"?`

Since the error recommends using --filter="- *", we can trim double quotes instead to permit that format.

Any fixes beyond that probably won't be trivial enough to fit in this PR, might even require an overhaul of the parser.

Reply:
Well, the parser issue runs deeper it seems. The above problem was in code, testing. Playing with the actual CLI, I found that it still regards `--filter="- *"` as two distinct flags and requires an additional set of quotes, e.g., 
`'--filter "- *"'`, `--filter="\"- *\""`, `-f '"- *"'`. Still not 1:1 with rsync.

I've tried every Kong tag to try and get this through but the parser refuses to read `-` as a simple string. Reading through some seemingly related issues marked as closed on the repo, it looks like the maintainer is set on preserving the behavior around this character (`-`) but maybe we can try to persuade him?

Reply:
That's surprising... but you're right, it can be easily reproduced with any argument, e.g. `--exodus-conf -foo.conf` crashes without even trying to read config from `-foo.conf`.

I don't think kong should behave like that... but I spent a while poking around the source and it was hard to judge which behavior in this area is intentional and which isn't. Anyway, it turns out that the kong API indeed does provide a way to avoid this: https://github.com/rohanpm/exodus-rsync/commit/0605a75fc941aa11b244b9ee76946faf2e6fea69

That commit is not complete, at least I didn't add a test for the error case.  Do you want to incorporate that workaround into your PR maybe?

Reply:
Ah, thanks for that, @rohanpm. I've included the workaround and added a test for it.